commands/start.js

```diff
@@ -23,32 +26,55 @@ function main(urc) {
   logger.log(
     `Starting underreact in ${urc.mode} mode. ${chalk.yellow('Wait ...')}`
   );
-  return (
-    del(urc.outputDirectory, { force: true })
-      // webpack needs to run first as others depend on the assets
-      .then(() => new Promise(resolve => watchWebpack(urc, resolve)))
-      .then(() => {
-        Promise.all([
-          writeHtml(urc),
-          writeCss(urc),
-          autoCopy.copy({
-            sourceDir: urc.publicDirectory,
-            destDir: urc.outputDirectory
-          })
-        ]);
-      })
-      .then(() => startServer(urc))
-      .then(() => {
-        watchCss(urc);
-        watchPublicDir(urc);
-      })
+
+  return del(urc.outputDirectory, { force: true }).then(() =>
+    watch(urc, () => startServer(urc))
   );
 }
 
-function watchWebpack(urc, onFirstRun) {
+function watch(urc, onFirstWrite) {
+  let cssOutput;
+  let webpackAssets;
+  let isFirst = true;
+  const htmlCachedWrite = writeHtml(urc);
```

Comment:
The function name `writeHtml` is confusing, because it's not actually *writing HTML*. Instead, it's creating a function that you can use to write HTML — right?

Reply:
Great point! I have renamed it to `htmlCompiler`.